hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockStoreProxy.java

```diff
@@ -33,83 +34,95 @@ public LockStoreProxy(LockStoreContainer container, ObjectNamespace namespace) {
 
     @Override
     public boolean lock(Data key, String caller, long threadId, long ttl) {
-        LockStore lockStore = getLockStore();
-        return lockStore.lock(key, caller, threadId, ttl);
+        LockStore lockStore = getLockStoreOrNull();
```

Comment:
- During Hazelcast.shutdownAll() process, LockService is shutted down before MapService and all lock stores are removed. This may cause NPE if there is an isLocked check in RecordStore. 

- Why is there a need to check `isLocked` when shutting down? 
That method is called also for `IMap#flush`.